quran/fields.py

```diff
@@ -0,0 +1,82 @@
+from django.core.exceptions import ValidationError
+from django.db import models
+
+
+class SegmentsField(models.TextField):
+    description = "Segments"
+
+    def to_python(self, value):
+        if value is None:
+            return None
+        # if value is empty string return empty list
+        if value == '':
+            return []
+        # if value is list
+        if isinstance(value, list):
+            # if list is not empty
+            if value:
+                # validate list of tuples
+                self._validate_tuples_containing_in_list(value)
+                self._validate_length_of_segments(
+                    value, "Length of tuples must be equals 2")
+            return value
+        try:
+            # try to parse a string of segments from DB to a list of tuples
```

Comment:
There is no DB in this function whatsoever ;)
```suggestion
            # try to convert a string of segments to a list of tuples
```